### PR TITLE
Mappings update trigger a reindex

### PR DESCRIPTION
**Pull request by clementbolin**
## What does this PR do ?

reindex documents with the new mappings (`collection:update`)

### How should this be manually tested?

Add a few documents in it.
Then add mappings to describe the fields in the added documents.

Expected: running a document:search in the collection should return the searched documents.

or run the functional tests

### CODE REVIEW

File: features/CollectionController.feature [hunk 1]
```diff
@@ -38,6 +38,29 @@ Feature: Collection Controller
     Then I should receive a "hits" array of objects matching:
       | _id          | _source      |
       | "document-1" | { "age": 2 } |
+  
+  Scenario: Update a collection and search document (dyanmic false)
```
Comment (Shiranuit) [inline features/CollectionController.feature:42]:
```suggestion
  Scenario: Update a collection and search document (dynamic false)
```

File: lib/service/storage/elasticsearch.js [hunk 2]
```diff
@@ -1387,6 +1387,20 @@ class ElasticSearch extends Service {
 
         await this.updateMapping(index, collection, mappings);
 
+        // update documents with new mappings
+        if (previousMappings.dynamic === 'false' && (await this._client.search(esRequest)).body.hits.hits.length > 0) {
```
Comment (Shiranuit) [inline lib/service/storage/elasticsearch.js:1391]: If think it's better to call `count` than `search` if you just want to know if there are documents to update

Side note:
Avoid putting long calls with chained property access like this in condition for clarity `await this._client.search(esRequest)).body.hits.hits.length` prefer storing the result in a const variable.
Conditions that are too long and uses `&&` or `||` should be splitted on multiple lines like so
```js
const documentsCount = await this._client.search(esRequest)).body.hits.hits.length;
if ( previousMappings.dynamic === 'false'
  && documentsCount > 0
) {
```

File: features/CollectionController.feature [hunk 3]
```diff
@@ -38,6 +38,29 @@ Feature: Collection Controller
...
+    When I "update" the collection "nyc-open-data":"green-taxi" with:
+      | mappings | { "properties": { "age": { "type": "long" } } } |
+    When I search documents with the following query:
+      """
+      {
+        "match": {
+          "age": 2
+        }
+      }
+      """
```
Comment (Shiranuit) [inline features/CollectionController.feature:59]: It's good but you also need to verify that the search does not match the field `age` before updating the mapping and triggering the reindex

**Review by Shiranuit — CHANGES_REQUESTED** [review]

**Comment by Aschen** [comment]
Stale PR

**Pull request closed without merging**